lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -216,17 +216,33 @@ in the response if any changes to the database were made.
 sub update {
     my $self = shift;
 
+    my $validation = $self->validation;
+    $validation->required('template');
+    $validation->required('schema')->like(qr/^[^.\/]+\.yaml$/);
+    $validation->optional('preview')->num(undef, 1);
+    $validation->optional('expand')->num(undef, 1);
+    $validation->optional('reference');
+    return $self->render(json => {error => ["No $_ specified"]}, status => 400) for @{$validation->failed};
+
+    if ($self->app->log->level eq 'debug') {
+        my @known_params = qw(template schema preview expand reference);
+        # Note: Account for built-in optional arguments supported by the Mojolicious framework
+        push @known_params, qw(format _method csrf_token);
```

Comment:
It would be good if the Mojolicious framework provided a way to query these.

Reply:
I should provide some context here. I got to pick @kraih 's brain a bit. There is no way right now. There might be in the future. This is why I'm proposing this for debug mode only. I would find it very valuable to have, even with that caveat.

Reply:
The framework will not help you with this. Like i said before, this is not supposed to be done and wrong.

Reply:
Okay, it seems I misunderstood your meaning. Thanks for correcting me.